### PR TITLE
RAINCATCH-915 - integration tests

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Tried to verify module myself by writing integration tests. 
Minor adjustments made inside module itself. 
Added integration tests (we will need to improve them a little bit, after some of the additional comments will be resolved) 

ping @paolobueno

### CODE REVIEW

**Comment by wtrocki** [comment]
Getting error "Task list must be set before calling createInstance" - investigating.
I think that we may need to enforce that on constructor?

File: cloud/wfm/src/executor/Executor.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import Process from '../process/Process';
  *
  * Triggers instantiation and execution of a {@link ProcessInstance}
  */
-interface Executor {
+export interface Executor {
```
Comment (paolobueno) [inline cloud/wfm/src/executor/Executor.ts:9]: These are already the default exports. any reason for exporting twice per file?

```javascript
import Executor, {Executor as Executor2} from './Executor';
assert(Executor === Executor2);
```

File: cloud/wfm/integration/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const executor: Executor = new ExecutorImpl(exampleProcess, repository);
+
+// Actual processing
+// tslint:disable-next-line:no-console
+console.log('[Integration] Starting process');
+// TODO explanation this looks like execution is async here.
+executor.start();
+//
```
Comment (paolobueno) [inline cloud/wfm/integration/index.ts:46]: Ah, very nice skeleton. The unit tests for each entity already involves using the others, though.

I'm not sure the term 'integration' testing makes sense within a single module, even though there are many moving pieces.

I was thinking about having these along with sync + a web api

**Comment by paolobueno** [comment]
@wtrocki about the `createInstance()` error yeah, maybe we could change the `ProcessImpl` constructor to take in the list of tasks already.

If you look at the unit tests the current setup is as follows:
https://github.com/feedhenry-raincatcher/raincatcher-core/blob/RAINCATCH-915-wfm/cloud/wfm/test/Process/index.ts#L11

File: cloud/wfm/src/executor/Executor.ts (hunk 1 above)
Comment (wtrocki) [reply cloud/wfm/src/executor/Executor.ts:9]: As per typesript guideline default exports should be avoided. Always export stuff normally to allow * import. defaults aren't imported with star interface. See other PRs

File: cloud/wfm/integration/index.ts (hunk 2 above)
Comment (wtrocki) [reply cloud/wfm/integration/index.ts:46]: That was something that was asked to be done.
In fact for this module we will need to do tests with couple other modules.. What name you propose?

**Comment by wtrocki** [comment]
Good idea with moving this to constructor. Setting tasks is not intuitive. I'm also not sure how to continue this. Executor has no methods to move things around etc.

File: cloud/wfm/src/executor/Executor.ts (hunk 1 above)
Comment (paolobueno) [reply cloud/wfm/src/executor/Executor.ts:9]: I see, we can merge this as is and remove all the default exports instead :+1:

**Pull request merged**